### PR TITLE
Subgraph apollo setup

### PR DESCRIPTION
**Pull request by Kenny-Gin1**
Closes: https://github.com/DXgovernance/davi-monorepo/issues/95 and https://github.com/DXgovernance/davi-monorepo/issues/119

Description: 
Provides an initial setup to support Subgraph in the front end by using the Apollo Client and querying data on multi-chains. 

- Wrapped the app with the `ApolloProvider`
- Implemented the `useGuildConfig` hook
- Dynamically switching networks depending on `chainId`
- Refactored the `wagmi` and the `apollo` clients into their separate folder
- Updated README
- Fixed an issue with the `davi` building with `yarn` instead of `pnpm`

NOTE: There's an `any` used in the `wagmi` client, and it seems to be a deeper issue of symlinking when doing monorepos. Did not find an easy fix for it.

### CODE REVIEW

File: apps/davi/src/clients/apollo.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  ),
+  // testnets
+  [SUPPORTED_DAVI_NETWORKS.GOERLI_ID]: setupApolloClient(
+    SUPPORTED_DAVI_NETWORKS.GOERLI_ID
+  ),
+  [SUPPORTED_DAVI_NETWORKS.ARBITRUM_TESTNET_ID]: setupApolloClient(
+    SUPPORTED_DAVI_NETWORKS.ARBITRUM_TESTNET_ID
+  ),
+  [SUPPORTED_DAVI_NETWORKS.LOCALHOST_ID]: setupApolloClient(
+    SUPPORTED_DAVI_NETWORKS.LOCALHOST_ID
```
Comment (rossneilson) [inline apps/davi/src/clients/apollo.ts:45]: Dont we need to add localhost here to test?

Comment (rossneilson) [reply apps/davi/src/clients/apollo.ts:45]: Meant to make this comment on line 16

**Comment by rossneilson** [comment]
Build error also

**Comment by Kenny-Gin1** [comment]
@rossneilson @dcrescimbeni Please review again. Changes: 
- Added `.graphclient`
- Added localhost URI (think it already worked before since we're also adding it in the `graphclient.yml) but just in case`

The expected behaviour right now is that it can only fetch and make queries to the Subgraph in localhost since that is the only place we have deployed the subgraph.

File: apps/davi/src/stores/modules/SnapshotRepGuild/index.ts [hunk 2]
```diff
@@ -53,7 +54,7 @@ export const snapshotRepGuildImplementation: Readonly<FullGovernanceImplementati
           useProposalCalls,
           useVotingResults,
           useVotingPowerOf,
-          useGuildConfig,
+          useGuildConfig: useGuildConfigFromSubgraph,
```
Comment (rossneilson) [inline apps/davi/src/stores/modules/SnapshotRepGuild/index.ts:57]: This should be duplicated in non rep guilds also

Comment (rossneilson) [reply apps/davi/src/stores/modules/SnapshotRepGuild/index.ts:57]: Also when added manually in there I seem to get 404s in the network tab

File: apps/davi/src/stores/modules/subgraph/common/useGuildConfig/index.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
+import { BigNumber } from 'ethers';
```
Comment (rulfo71) [inline apps/davi/src/stores/modules/subgraph/common/useGuildConfig/index.ts:1]: should we have this on a useGuildConfig.ts and export it from index?

File: apps/davi/src/stores/modules/SnapshotRepGuild/index.ts (hunk 2 above)
Comment (Kenny-Gin1) [reply apps/davi/src/stores/modules/SnapshotRepGuild/index.ts:57]: > Also when added manually in there I seem to get 404s in the network tab

Where does this happen and how to reproduce this?

File: apps/davi/src/stores/modules/subgraph/common/useGuildConfig/index.ts (hunk 3 above)
Comment (Kenny-Gin1) [reply apps/davi/src/stores/modules/subgraph/common/useGuildConfig/index.ts:1]: Could you elaborate on this? Do you mean that the folder structure should look different?

**Comment by rulfo71** [comment]
@Kenny-Gin1 this should be going through useGuildConfigFromSubgraph right? in my case is going through the useGuildConfig from ../common/fetchers

**Comment by dcrescimbeni** [comment]
It's working for me now :+1:

File: apps/davi/src/stores/modules/SnapshotRepGuild/index.ts (hunk 2 above)
Comment (rossneilson) [reply apps/davi/src/stores/modules/SnapshotRepGuild/index.ts:57]: Fixed with the uri update

File: apps/davi/src/stores/modules/subgraph/common/useGuildConfig/index.ts (hunk 3 above)
Comment (rulfo71) [reply apps/davi/src/stores/modules/subgraph/common/useGuildConfig/index.ts:1]: yes, i mean having this: 

+ subgraph
|-+ common
| |-+ useGuildConfig 
| | |- index.ts
| | |- useGuildConfig.ts
| | | |- query.graphql

let me know if i'm not being clear

**Comment by Kenny-Gin1** [comment]
Great to hear @rossneilson @dcrescimbeni! Thanks for the feedback!

**Comment by Kenny-Gin1** [comment]
> @Kenny-Gin1 this should be going through useGuildConfigFromSubgraph right? in my case is going through the useGuildConfig from ../common/fetchers

This is solved now!

**Pull request merged**